datafusion/proto/tests/cases/roundtrip_physical_plan.rs

```diff
@@ -1736,3 +1737,57 @@ async fn roundtrip_physical_plan_node() {
 
     let _ = plan.execute(0, ctx.task_ctx()).unwrap();
 }
+
+#[tokio::test]
+async fn test_tpch_part_in_list_query_with_real_parquet_data() -> Result<()> {
+    // Test the specific query: SELECT p_size FROM part WHERE p_size IN (14, 6, 5, 31)
+    // This uses REAL TPC-H parquet data to reproduce the serialization bug with actual data sources
+    //
+    // NOTE: This test requires TPC-H data to reproduce the serialization bug.
+    // Using alltypes_plain.parquet does NOT reproduce the issue, suggesting the bug
+    // is specific to certain characteristics of TPC-H parquet files or their schema.
```

Comment:
I have tested on data for scale factors 1 and 2; and hit the same problem. 

If you do not know how to generate tpch data, use [tpchgen-rs](https://github.com/clflushopt/tpchgen-rs)